docs/new-feature-tutorial.rst

```diff
@@ -2,84 +2,178 @@
+
+  def apply_events(state, events, user_profile):
+    for event in events:
+      # ...
+      elif event['type'] == 'realm':
+         field = 'realm_' + event['property']
+         state[field] = event['value']
+
+This feature adds a new parameter that should be sent to clients when the
+application loads and be accessible via JavaScript, so add it the initial state
+in ``zerver/views/__init__.py``. ::
+
+  # zerver/views/__init__.py
+
+  def home(request):
+    # ...
+    page_params = dict(
+      # ...
+      realm_invite_by_admins_only = register_ret['realm_invite_by_admins_only'],
+      # ...
+    )
+
+Since this feature also adds a checkbox to the admin page, and adds a new property the Realm model that can be modified from there, you also need to make changes to the ``update_realm`` function in the same file: ::
+
+  # zerver/views/__init__.py
+
+  def update_realm(request, user_profile,
+    name=REQ(validator=check_string, default=None),
+    restricted_to_domain=REQ(validator=check_bool, default=None),
+    invite_by_admins_only=REQ(validator=check_bool,default=None)):
+
+    # ...
+
+    if invite_by_admins_only is not None and
+      realm.invite_by_admins_only != invite_by_admins_only:
+        do_set_realm_invite_by_admins_only(realm, invite_by_admins_only)
+        data['invite_by_admins_only'] = invite_by_admins_only
+
+Then make the required front end changes: in this case a checkbox needs to be added to the admin page (and its value added to the data sent back to server when a realm is updated) and the change event needs to be handled on the client.
+
+To add the checkbox to the admin page, modify the relevant template, ``static/templates/admin_tab.handlebars`` (omitted here since it is relatively straightforward). Then add code to handle changes to the new form control in ``static/js/admin.js``. ::
+
+  var url = "/json/realm";
+  var new_invite_by_admins_only =
+    $("#id_realm_invite_by_admins_only").prop("checked");
+  data[invite_by_admins_only] = JSON.stringify(new_invite_by_admins_only);
+
+  channel.patch({
+    url: url,
+    data: data,
+    success: function (data) {
+      # ...
+      if (data.invite_by_admins_only) {
+        ui.report_success("New users must be invited by an admin!", invite_by_admins_only_status);
+      } else {
+        ui.report_success("Any user may now invite new users!", invite_by_admins_only_status);
+      }
+      # ...
+    }
+  });
+
+Finally, update ``server_events.js`` to handle related events coming from the server. ::
+
+  # static/js/server_events.js
+
+  function get_events_success(events) {
+    # ...
+    var dispatch_event = function dispatch_event(event) {
+        switch (event.type) {
+        # ...
+        case 'realm':
+          if (event.op === 'update' && event.property === 'invite_by_admins_only') {
+            page_params.realm_invite_by_admins_only = event.value;
+        }
+    }
+  }
```

Comment:
Maybe at the end, link to the entire commit in the Git history so that they can review?  If we do this, we should mention that Zulip was upgraded from Django 1.6 => 1.8 since then and thus the generated migrations format has changed.